In [18]:
# 2.1 Carga inicial y exploración básica del dataset

import pandas as pd

# Cargar dataset desde la carpeta raw
df_original = pd.read_csv("../1_data/raw/WA_Fn-UseC_-HR-Employee-Attrition.csv")

# Fuente del dataset
fuente = "IBM HR Analytics Employee Attrition & Performance Dataset - Kaggle"
print("Fuente ->", fuente)

# Mostrar primeras filas
df_original.head()

Fuente -> IBM HR Analytics Employee Attrition & Performance Dataset - Kaggle


,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2


In [19]:
# Hago un info para ver el tipo de datos que tiene el dataset
df_original.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Age                       1470 non-null   int64 
 1   Attrition                 1470 non-null   object
 2   BusinessTravel            1470 non-null   object
 3   DailyRate                 1470 non-null   int64 
 4   Department                1470 non-null   object
 5   DistanceFromHome          1470 non-null   int64 
 6   Education                 1470 non-null   int64 
 7   EducationField            1470 non-null   object
 8   EmployeeCount             1470 non-null   int64 
 9   EmployeeNumber            1470 non-null   int64 
 10  EnvironmentSatisfaction   1470 non-null   int64 
 11  Gender                    1470 non-null   object
 12  HourlyRate                1470 non-null   int64 
 13  JobInvolvement            1470 non-null   int64 
 14  JobLevel                

In [ ]:
# Reviso de qué dimensiones estoy hablando

# Número de filas y columnas
df_original.shape

(1470, 35)

In [ ]:
# Revisar de qué tipo de datos estoy hablando

# Tipos de datos por columna
df_original.dtypes

Age                          int64
Attrition                   object
BusinessTravel              object
DailyRate                    int64
Department                  object
DistanceFromHome             int64
Education                    int64
EducationField              object
EmployeeCount                int64
EmployeeNumber               int64
EnvironmentSatisfaction      int64
Gender                      object
HourlyRate                   int64
JobInvolvement               int64
JobLevel                     int64
JobRole                     object
JobSatisfaction              int64
MaritalStatus               object
MonthlyIncome                int64
MonthlyRate                  int64
NumCompaniesWorked           int64
Over18                      object
OverTime                    object
PercentSalaryHike            int64
PerformanceRating            int64
RelationshipSatisfaction     int64
StandardHours                int64
StockOptionLevel             int64
TotalWorkingYears   

In [ ]:
# Resumen estadístico
df_original.describe().T


,count,mean,std,min,25%,50%,75%,max
Age,1470.0,36.923810,9.135373,18.0,30.00,36.0,43.00,60.0
DailyRate,1470.0,802.485714,403.509100,102.0,465.00,802.0,1157.00,1499.0
DistanceFromHome,1470.0,9.192517,8.106864,1.0,2.00,7.0,14.00,29.0
Education,1470.0,2.912925,1.024165,1.0,2.00,3.0,4.00,5.0
EmployeeCount,1470.0,1.000000,0.000000,1.0,1.00,1.0,1.00,1.0
EmployeeNumber,1470.0,1024.865306,602.024335,1.0,491.25,1020.5,1555.75,2068.0
EnvironmentSatisfaction,1470.0,2.721769,1.093082,1.0,2.00,3.0,4.00,4.0
HourlyRate,1470.0,65.891156,20.329428,30.0,48.00,66.0,83.75,100.0
JobInvolvement,1470.0,2.729932,0.711561,1.0,2.00,3.0,3.00,4.0
JobLevel,1470.0,2.063946,1.106940,1.0,1.00,2.0,3.00,5.0


In [ ]:
# Ver cuántas variables categóricas hay
categoricas = df_original.select_dtypes(include=['object']).columns
categoricas



Index(['Attrition', 'BusinessTravel', 'Department', 'EducationField', 'Gender',
       'JobRole', 'MaritalStatus', 'Over18', 'OverTime'],
      dtype='object')

In [ ]:
# Análisis de cada variable categórica
for col in categoricas:
    print(f"🔹 {col}:")
    
    # Cuántos valores únicos tiene
    unicos = df_original[col].unique()
    print(f"   Valores posibles: {list(unicos)}")
    
    # Cuántos casos de cada valor
    conteo = df_original[col].value_counts()
    print(f"   Cantidad de cada valor:")
    
    for valor, cantidad in conteo.items():
        porcentaje = (cantidad / len(df_original)) * 100
        print(f"   • {valor}: {cantidad} personas ({porcentaje:.1f}%)")
    
    print("-" * 40)


🔹 Attrition:
   Valores posibles: ['Yes', 'No']
   Cantidad de cada valor:
   • No: 1233 personas (83.9%)
   • Yes: 237 personas (16.1%)
----------------------------------------
🔹 BusinessTravel:
   Valores posibles: ['Travel_Rarely', 'Travel_Frequently', 'Non-Travel']
   Cantidad de cada valor:
   • Travel_Rarely: 1043 personas (71.0%)
   • Travel_Frequently: 277 personas (18.8%)
   • Non-Travel: 150 personas (10.2%)
----------------------------------------
🔹 Department:
   Valores posibles: ['Sales', 'Research & Development', 'Human Resources']
   Cantidad de cada valor:
   • Research & Development: 961 personas (65.4%)
   • Sales: 446 personas (30.3%)
   • Human Resources: 63 personas (4.3%)
----------------------------------------
🔹 EducationField:
   Valores posibles: ['Life Sciences', 'Other', 'Medical', 'Marketing', 'Technical Degree', 'Human Resources']
   Cantidad de cada valor:
   • Life Sciences: 606 personas (41.2%)
   • Medical: 464 personas (31.6%)
   • Marketing: 159 per

In [ ]:
# Análisis Attrition 
print("\n ANÁLISIS DE ROTACIÓN (ATTRITION)")
print("-" * 40)

attrition_counts = df_original['Attrition'].value_counts()
print(f"Personas que NO se fueron: {attrition_counts.get('No', 0)}")
print(f"Personas que SÍ se fueron: {attrition_counts.get('Yes', 0)}")

tasa_rotacion = (attrition_counts.get('Yes', 0) / len(df_original)) * 100
print(f" Tasa de rotación: {tasa_rotacion:.1f}%")
print(f" Por cada 100 personas, {tasa_rotacion:.0f} se van")

# Ver cómo afecta OverTime a la rotación 
print("\n ¿AFECTAN LAS HORAS EXTRAS A LA ROTACIÓN?")
print("-" * 40)

# Crear una tabla simple
tabla_overtime = df_original.groupby('OverTime')['Attrition'].value_counts().unstack()
print(tabla_overtime)

# Calcular porcentajes
print("\nPorcentajes por grupo:")
for overtime in ['Yes', 'No']:
    total = tabla_overtime.loc[overtime].sum()
    se_fue = tabla_overtime.loc[overtime, 'Yes'] if 'Yes' in tabla_overtime.columns else 0
    porcentaje = (se_fue / total) * 100 if total > 0 else 0
    
    print(f"• Con horas extras ({overtime}): {porcentaje:.1f}% se van")

print("=" * 50)
print(" Análisis de variables categóricas completado")


 ANÁLISIS DE ROTACIÓN (ATTRITION)
----------------------------------------
Personas que NO se fueron: 1233
Personas que SÍ se fueron: 237
 Tasa de rotación: 16.1%
 Por cada 100 personas, 16 se van

 ¿AFECTAN LAS HORAS EXTRAS A LA ROTACIÓN?
----------------------------------------
Attrition   No  Yes
OverTime           
No         944  110
Yes        289  127

Porcentajes por grupo:
• Con horas extras (Yes): 30.5% se van
• Con horas extras (No): 10.4% se van
 Análisis de variables categóricas completado


### Lo que he aprendido hasta ahora:

**Sobre el dataset:**
- Tengo datos de 1,470 personas que trabajan en una empresa
- Cada persona tiene 35 características (edad, salario, satisfacción, etc.)
- No faltan datos, todo está completo
- **Variable importante**: `Attrition` (si la persona se quedó o se fue)

**Lo más interesante que he descubierto:**
De las 1,470 personas:
- 1,233 se quedaron en la empresa 
- 237 se fueron (16.1%)

**Un hallazgo importante:**
- 30 de cada 100 personas que hacen horas extras (`OverTime = "Yes"`) se van
- Solo 10 de cada 100 personas que no hacen horas extras se van
* Hacer horas extras triplica la probabilidad de irse.

**Variables que voy a estudiar más:**
1. `Attrition` - ¿Quién se queda y quién se va?
2. `JobSatisfaction` - ¿Cómo están de satisfechas/os con su trabajo?
3. `OverTime` - ¿Hacen horas extras?
4. `WorkLifeBalance` - ¿Tienen equilibrio vida-trabajo?
5. `YearsAtCompany` - ¿Cuántos años llevan en la empresa?

**¿Por qué estas variables?**
Porque mi hipótesis es que:
- Las personas más satisfechas se quedan más tiempo
- Las personas con buen equilibrio vida-trabajo son más felices
- Las horas extras pueden cansar a las personas y hacer que se vayan

**Lo siguiente que haré:**
En el próximo notebook voy a limpiar los datos:
1. Quitar columnas que no uso (como número de empleado)
2. Convertir "Sí/No" a números (1 para Sí, 0 para No)
3. Ver si hay datos raros o errores